### ISSUE TITLE
UnusedLocalVariable false positive for use()

### ISSUE THREAD

**Issue by zhoffman-ku** [phpmd/phpmd#325]
Received message:

```
    Avoid unused local variables such as '$factory'.
```

Code:

```
$factory = $builder->getFormFactory(); 
$builder->addEventListener(FormEvents::PRE_SET_DATA, function(FormEvent $event) use ($factory){
    /*...*/
});
```

Using nb-phpmd-phpcs-phpcpd-checkstyle-20131204-nb7.4.201310111528.nbm with NetBeans 8.1.

**Comment by ravage84** [comment]
``20131204`` sounds to me like 2013-12-04, which would be quite an old version.
Can you confirm this with something newer?
Or better said, can you check your code with a vanilla, up to date version of PHPMD?

**Comment by rocketlatotzky** [comment]
confirmed in 2.3.2

**Comment by poisa** [comment]
I was about to report this also when I realized that you have to aactually *do* something with the variable inside the closure for phpmd to remove the warning. `use()`ing it alone won't do. For example:

```php
<?php

namespace MyNamespace;

class Test
{
    public function test()
    {
        $array = [1, 2, 3];
        $data = 'foo';

        array_map(function ($item) use ($data) {
            print_r($item);
            print_r($data); // comment out this line and phpmd will mark $data
                            // as unused OUTSIDE the closure (which I think is correct)
        }, $array);
    }
}
```

**Comment by ravage84** [comment]
Thank you very much @poisa.

@zhoffman-ku & @rocketlatotzky I'm going to close this. If you have a code example that contradicts with what @poisa reported, we can re-open the issue.
